api/src/main/scala/uk/ac/wellcome/platform/api/utils/ApiRequestUtils.scala

```diff
@@ -11,9 +11,21 @@ object ApiRequestUtils {
       case _ => "http"
     }
 
-    val hostHeader = request.host match {
-      case Some(host) => host
-      case _ => throw new BadRequestException("Host header not set")
+    // Most users will access our API through a DNS record that comes through
+    // https://api.wellcomecollection.org.  I can't find a way to detect
+    // the DNS name through the request headers, so just hard-code it if we
+    // detect the request came through ELB.
+    //
+    // Specifically, we look for a header that is added to all ELB requests:
+    // http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html
+    val hostHeader: String = request.headerMap.get("X-Amzn-Trace-Id") match {
```

Comment:
Might it make more sense to have the DNS name in config (and defaulted to localhost)?